src/flags/color.rs

```diff
@@ -23,7 +26,73 @@ impl Color {
     /// The [ColorOption] is configured with their respective [Configurable] implementation.
     pub fn configure_from(matches: &ArgMatches, config: &Config) -> Self {
         let when = ColorOption::configure_from(matches, config);
-        Self { when }
+        let theme = ThemeOption::from_config(config);
+        Self { when, theme }
+    }
+}
+
+/// ThemeOption could be one of the following:
+/// Custom(*.yaml): use the YAML theme file as theme file
+/// if error happened, use the default theme
+#[derive(PartialEq, Eq, Debug, Clone)]
+pub enum ThemeOption {
+    NoColor,
+    Default,
+    NoLscolors,
+    Custom(String),
+}
+
+impl ThemeOption {
+    fn from_config(config: &Config) -> ThemeOption {
+        if let Some(classic) = config.classic {
+            if classic {
+                return ThemeOption::NoColor;
+            }
+        }
+        if let Some(c) = &config.color {
+            if let Some(t) = &c.theme {
+                return t.clone();
+            }
+        }
+
+        ThemeOption::default()
+    }
+}
+
+impl Default for ThemeOption {
+    fn default() -> Self {
+        ThemeOption::Default
+    }
+}
+
+impl<'de> de::Deserialize<'de> for ThemeOption {
+    fn deserialize<D>(deserializer: D) -> Result<ThemeOption, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct ThemeOptionVisitor;
+
+        impl<'de> Visitor<'de> for ThemeOptionVisitor {
+            type Value = ThemeOption;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("`no-color`, `default`, `no-lscolors` or <theme-file-path>")
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<ThemeOption, E>
+            where
+                E: de::Error,
+            {
+                match value {
+                    "no-color" => Ok(ThemeOption::NoColor),
+                    "default" => Ok(ThemeOption::Default),
+                    "no-lscolors" => Ok(ThemeOption::NoLscolors),
```

Comment:
This can be simplified. If the user has not mentioned any, just don't do anything and the `Default` will be used. If the user has mentioned something we do a file lookup.

This would mean that if the user mentions `default` we will be looking up the `default.yaml` file and erroring out if it does not exist. This might prevent us from packaging themes with `lsd` binary but other than that it might be a better option. What do you think?

Reply:
I would like to keep the current implementation, because,

1. the `default.yaml` disk io is not necessary
2. it may be necessary for us to package default dark and default white theme with `lsd`, even we support theme config now.

Reply:
It might end up being useful for dark and light if we end up shipping it in future. I guess we can keep the code around but just drop `no-color` and `no-lscolors` for now.
We can drop `no-color` as it is exactly the same as `color: never` and is technically not a theme. As for `no-lscolors`, that is also not really a theme, but an option to disable LSCOLORS in the default theme.

Reply:
the `ThemeOption::default` and `ThemeOption::NoLscolor` are originally used by `lsd`, which controls most of coloring logic.

there is quite a lot of work to change this logic here, I have deleted these two options in the readme, maybe we should just leave this implementation here and improve it later

Reply:
What I meant by remove those is to just not parse the string. No need to change anything in the enum. For example if the user writes `no-color`, instead of assigning to `ThemeOption::default` we should look up `themes/no-color.yaml`.